allennlp/data/data_loaders/multitask_data_loader.py

```diff
@@ -309,9 +247,9 @@ def _get_instances_for_epoch(self) -> Dict[str, Iterable[Instance]]:
 
     def _make_data_loader(self, key: str) -> MultiProcessDataLoader:
         kwargs: Dict[str, Any] = {}
-        kwargs["reader"] = self.readers[key]
+        kwargs["reader"] = _MultitaskDatasetReaderShim(self.readers[key], key)
         kwargs["data_path"] = self.data_paths[key]
-        kwargs["batch_size"] = 1
+        kwargs["batch_size"] = 1  # So that the loader gives us one instance at a time.
```

Comment:
We only use it to call `iter_instances()`, right? So `batch_size` won't matter.

Reply:
True, but we have to set something. I clarified the comment.